odc/stac/_aws.py

```diff
@@ -5,16 +5,25 @@
 """
```

Comment:
What is your preferred behavior? Remove `get_aws_settings` from the public interface if `botocore` isn't available?

Reply:
it's not part of public interface though, `configure_s3_access` is. 

I think best way to deal with this is to keep `configure_s3_access`  public but raise error message if used without `botocore` installed, but do all that inside `_rio.py` keeping `_aws.py` as it is now, without any guards and meta programming that can interfere with types.

If you think it will make it clearer, feel free to rename every function inside _aws to start with `_`, to me these are "public functions within a private module" so not prefixed with `_`.

Reply:
These are public:

https://github.com/opendatacube/odc-stac/blob/1c6fd02b6a95f5ebca3da9fb3dfe4df210eae105/odc/stac/__init__.py#L16-L27

and also these when "datacube" is installed:

https://github.com/opendatacube/odc-stac/blob/1c6fd02b6a95f5ebca3da9fb3dfe4df210eae105/odc/stac/eo3/__init__.py#L9-L10

and finally "bench" tools, which need extra libs too

https://github.com/opendatacube/odc-stac/blob/1c6fd02b6a95f5ebca3da9fb3dfe4df210eae105/odc/stac/bench/__init__.py#L14-L22